diff --git a/README.md b/README.md
index 54bf543..3548de5 100644
--- a/README.md
+++ b/README.md
@@ -39,6 +39,8 @@ Or if you want to install all plugins provided, including the use of the `jpegph
 ./ipa-plugins-installer.sh --all --reload
 ```
 
+Note: the plugins are developed, tested and used with FreeIPA version 4.8.
+
 
 ## License
 
diff --git a/ipa-plugins-installer.sh b/ipa-plugins-installer.sh
index 8243d7a..bc725f6 100755
--- a/ipa-plugins-installer.sh
+++ b/ipa-plugins-installer.sh
@@ -3,12 +3,25 @@
 ###
 ### Copyright 2021 téïcée SARL <https://www.teicee.com>
 ### Written by Grégory MARIGOT <gmarigot at teicee.com>
+###
+### This program is free software: you can redistribute it and/or modify
+### it under the terms of the GNU Affero General Public License as
+### published by the Free Software Foundation, either version 3 of the
+### License, or (at your option) any later version.
+###
+### This program is distributed in the hope that it will be useful,
+### but WITHOUT ANY WARRANTY; without even the implied warranty of
+### MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+### GNU Affero General Public License for more details.
+###
+### You should have received a copy of the GNU Affero General Public License
+### along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 ### Variables
 DIR_SRC_PLUGINS="$(dirname $0)"
 DIR_IPA_JS="/usr/share/ipa/ui/js"
 DIR_IPA_PY="$(ls -1 -d /usr/lib/python*/*-packages/ipaserver/plugins 2>/dev/null |head -n1)"
-ALL_PLUGIN="$(ls -1 "${DIR_SRC_PLUGINS}/"*.js "${DIR_SRC_PLUGINS}/"*.py |sed 's|^.*/||' |sed 's/\.\(js\|py\)$//' |sort |uniq |tr '\n' ' ')"
+ALL_PLUGIN="$(ls -1 "${DIR_SRC_PLUGINS}/"*.js "${DIR_SRC_PLUGINS}/"*.py 2>/dev/null |sed 's|^.*/||' |sed 's/\.\(js\|py\)$//' |sort |uniq |tr '\n' ' ')"
 LST_PLUGIN=
 OPT_RELOAD=
 
diff --git a/user_jpegphoto.js b/user_jpegphoto.js
index 6b5a4e6..e5b182e 100644
--- a/user_jpegphoto.js
+++ b/user_jpegphoto.js
@@ -3,6 +3,19 @@
  *
  * Copyright 2021 téïcée SARL <https://www.teicee.com>
  * Written by Grégory MARIGOT <gmarigot at teicee.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 define(
 	[
diff --git a/user_jpegphoto.py b/user_jpegphoto.py
index bf6a154..e6d10e0 100644
--- a/user_jpegphoto.py
+++ b/user_jpegphoto.py
@@ -2,6 +2,19 @@
 ##
 ## Copyright 2021 téïcée SARL <https://www.teicee.com>
 ## Written by Grégory MARIGOT <gmarigot at teicee.com>
+##
+## This program is free software: you can redistribute it and/or modify
+## it under the terms of the GNU Affero General Public License as
+## published by the Free Software Foundation, either version 3 of the
+## License, or (at your option) any later version.
+##
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU Affero General Public License for more details.
+##
+## You should have received a copy of the GNU Affero General Public License
+## along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 from ipaserver.plugins.user import user
 from ipalib import Bytes, _
diff --git a/widget_picture.js b/widget_picture.js
index e21e72f..c2d05e6 100644
--- a/widget_picture.js
+++ b/widget_picture.js
@@ -3,6 +3,19 @@
  *
  * Copyright 2021 téïcée SARL <https://www.teicee.com>
  * Written by Grégory MARIGOT <gmarigot at teicee.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 define(
 	[
